src/control/lib/control/pool.go

```diff
@@ -1279,27 +1290,36 @@ func GetMaxPoolSize(ctx context.Context, log logging.Logger, rpcClient UnaryInvo
 					continue
 				}
 
+				if len(ranks) != 0 && !smdDevice.Rank.InList(ranks) {
+					log.Debugf("Skipping SMD device %s (instance %d, ctrlr %s, rank %s) not in ranklist: %s",
+						smdDevice.UUID, smdDevice.Rank, smdDevice.TrAddr, smdDevice.Rank, ranks)
+					continue
+				}
+
 				nvmeRanksBytes[smdDevice.Rank] += smdDevice.AvailBytes
-				log.Debugf("Adding SMD device %s (instance %d, ctrlr %s) is usable: "+
-					"device state=%q, size=%d total=%d",
+				log.Debugf("Adding SMD device %s (instance %d, ctrlr %s) is usable: device state=%q, size=%d total=%d",
 					smdDevice.UUID, smdDevice.Rank, smdDevice.TrAddr,
 					smdDevice.NvmeState.String(), smdDevice.AvailBytes,
 					nvmeRanksBytes[smdDevice.Rank])
 			}
 		}
-		for _, nvmeRankBytes := range nvmeRanksBytes {
+		for rank, nvmeRankBytes := range nvmeRanksBytes {
 			if nvmeBytes > nvmeRankBytes {
 				nvmeBytes = nvmeRankBytes
 			}
+			if _, ok := ranksWithoutSmd[rank]; ok {
+				delete(ranksWithoutSmd, rank)
+			}
 		}
 	}
 
 	if nvmeBytes == math.MaxUint64 {
 		nvmeBytes = 0
 	}
 
-	// TODO (DAOS-9557) Check if there is no ranks (i.e. rank with SCM available) with some NVMe
-	// storage and other without
+	if len(ranksWithoutSmd) != 0 {
+		nvmeBytes = 0
+	}
```

Comment:
I think it's worth keeping a comment here explaining why. This is sort of a weird scenario.

Reply:
- [ ] Add comments